lib/rubocop/cop/internal_affairs/redundant_message_argument.rb

```diff
@@ -10,39 +10,61 @@ module InternalAffairs
       # @example
       #
       #   # bad
-      #   add_offense(node, :expression, MSG)
-      #   add_offense(node, :expression, message)
-      #   add_offense(node, :expression, message(node))
+      #   add_offense(node, message: MSG)
+      #   add_offense(node, message: message)
+      #   add_offense(node, message: message(node))
       #
       #   # good
-      #   add_offense(node, :expression)
-      #   add_offense(node, :expression, CUSTOM_MSG)
-      #   add_offense(node, :expression, message(other_node))
+      #   add_offense(node)
+      #   add_offense(node, message: CUSTOM_MSG)
+      #   add_offense(node, message: message(other_node))
       #
       class RedundantMessageArgument < Cop
```

Comment:
Ugh. This implementation ended up with a lot of indirection, but I think our node pattern matchers currently can't match "any pair", so I might look into that. 🙂 For now, I think this implementation is okay.

Reply:
Can't agree more, we need "any pair" matchers desperately. 
If you're wondering, I have borrowed used pattern from `Rails/DelegateAllowBlank`:
https://github.com/bbatsov/rubocop/blob/7416bab77b9fd3c6f6476c71d5ecbd27910ec1a5/lib/rubocop/cop/rails/delegate_allow_blank.rb#L19-L25
https://github.com/bbatsov/rubocop/blob/7416bab77b9fd3c6f6476c71d5ecbd27910ec1a5/lib/rubocop/cop/rails/delegate_allow_blank.rb#L43-L47

I've looked inside `RuboCop::NodePattern` class, but didn't manage how to introduce the "any pair" matchers. Will invest some more time, though, will be happy if you or someone else will implement it earlier.

Reply:
@akhramov: Yeah. The node pattern parser takes a while to wrap your head around. 🙂 The more people who are familiar with it, the better!